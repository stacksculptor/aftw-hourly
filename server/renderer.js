const path = require('path');
const fs = require('fs');
const _ = require('lodash');
const { renderApp } = require('./importer');

const buildPath = path.resolve(__dirname, '..', 'build');

// The HTML build file is generated from the `public/index.html` file
// and used as a template for server side rendering. The application
// head and body are injected to the template from the results of
// calling the `renderApp` function imported from the bundle above.
const indexHtml = fs.readFileSync(path.join(buildPath, 'index.html'), 'utf-8');

const reNoMatch = /($^)/;
const template = _.template(indexHtml, {
  // Interpolate variables in the HTML template with the following
  // syntax: <!--!variableName-->
  //
  // This syntax is very intentional: it works as a HTML comment and
  // doesn't render anything visual in the dev mode, and in the
  // production mode, HtmlWebpackPlugin strips out comments using
  // HTMLMinifier except those that aren't explicitly marked as custom
  // comments. By default, custom comments are those that begin with a
  // ! character.
  //
  // Note that the variables are _not_ escaped since we only inject
  // HTML content.
  //
  // See:
  // - https://github.com/ampedandwired/html-webpack-plugin
  // - https://github.com/kangax/html-minifier
  // - Plugin options in the production Webpack configuration file
  interpolate: /<!--!([\s\S]+?)-->/g,
  // Disable evaluated and escaped variables in the template
  evaluate: reNoMatch,
  escape: reNoMatch,
});

exports.render = function(requestUrl, context, preloadedState) {
  const { head, body } = renderApp(requestUrl, context, preloadedState);

  // Preloaded state needs to be passed for client side too.
  // For security reasons we ensure that preloaded state is considered as a string
  // by replacing '<' character with its unicode equivalent.
  // http://redux.js.org/docs/recipes/ServerRendering.html#security-considerations
  const serializedState = JSON.stringify(preloadedState).replace(/</g, '\\u003c');
  const preloadedStateScript = `
      <script>window.__PRELOADED_STATE__ = ${serializedState};</script>
  `;

  return template({ title: head.title.toString(), preloadedStateScript, body });
};
