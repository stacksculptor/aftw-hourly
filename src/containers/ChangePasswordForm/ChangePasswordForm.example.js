/* eslint-disable no-console, import/prefer-default-export */
import ChangePasswordForm from './ChangePasswordForm';

export const Empty = {
  component: ChangePasswordForm,
  props: {
    onSubmit(values) {
      console.log('submit new password form values:', values);
    },
  },
};
