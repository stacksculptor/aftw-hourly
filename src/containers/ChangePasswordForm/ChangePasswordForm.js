import React from 'react';
import { Field, reduxForm, propTypes as formPropTypes } from 'redux-form';
import { Button } from '../../components';

const ChangePasswordForm = props => {
  const { handleSubmit, pristine, submitting } = props;
  return (
    <form onSubmit={handleSubmit}>
      <label htmlFor="newPassword1">New password</label>
      <Field name="newPassword1" component="input" type="password" />
      <label htmlFor="newPassword2">New password, again</label>
      <Field name="newPassword2" component="input" type="password" />
      <Button type="submit" disabled={pristine || submitting}>Change password</Button>
    </form>
  );
};

ChangePasswordForm.propTypes = { ...formPropTypes };

export default reduxForm({ form: 'changePassword' })(ChangePasswordForm);
