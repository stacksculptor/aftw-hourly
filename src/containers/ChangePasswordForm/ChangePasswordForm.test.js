import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from '../../store';
import ChangePasswordForm from './ChangePasswordForm';

describe('ChangePasswordForm', () => {
  it('matches snapshot', () => {
    const store = configureStore();
    const component = renderer.create(
      (
        <Provider store={store}>
          <ChangePasswordForm />
        </Provider>
      ),
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
