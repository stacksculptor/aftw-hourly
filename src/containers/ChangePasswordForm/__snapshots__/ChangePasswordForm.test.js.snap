exports[`ChangePasswordForm matches snapshot 1`] = `
<form
  onSubmit={[Function]}>
  <label
    htmlFor="email">
    Email
  </label>
  <input
    name="email"
    onBlur={[Function]}
    onChange={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    onFocus={[Function]}
    type="email"
    value="" />
  <label
    htmlFor="newPassword1">
    New password
  </label>
  <input
    name="newPassword1"
    onBlur={[Function]}
    onChange={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    onFocus={[Function]}
    type="password"
    value="" />
  <label
    htmlFor="newPassword2">
    New password, again
  </label>
  <input
    name="newPassword2"
    onBlur={[Function]}
    onChange={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    onFocus={[Function]}
    type="password"
    value="" />
  <label
    htmlFor="password">
    Current password
  </label>
  <input
    name="password"
    onBlur={[Function]}
    onChange={[Function]}
    onDragStart={[Function]}
    onDrop={[Function]}
    onFocus={[Function]}
    type="password"
    value="" />
  <button
    disabled={true}
    type="submit">
    Change password
  </button>
</form>
`;
